vscode-dotnet-runtime-library/src/Utils/CommandExecutor.ts

```diff
@@ -379,35 +380,55 @@ ${(commandOutputJson as CommandExecutorResult).stderr}.`),
      *
      * @param workingDirectory The directory to execute in. Only works for non sudo commands.
      * @param terminalFailure Whether to throw up an error when executing under sudo or suppress it and return stderr
+     * @param options the dictionary of options to forward to the child_process. Set dotnetInstallToolCacheTtlMs=number to cache the result with a ttl and also use cached results.
      * @returns the result(s) of each command. Can throw generically if the command fails.
      */
     public async execute(command: CommandExecutorCommand, options: any = null, terminalFailure = true): Promise<CommandExecutorResult>
     {
         const fullCommandString = `${command.commandRoot} ${command.commandParts.join(' ')}`;
+        let useCache = false;
         // Remove this when https://github.com/typescript-eslint/typescript-eslint/issues/2728 is done
         // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
-        if (options && !options?.cwd)
+        if (options)
         {
-            // Remove this when https://github.com/typescript-eslint/typescript-eslint/issues/2728 is done
             // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
-            options.cwd = path.resolve(__dirname);
-        }
-        // Remove this when https://github.com/typescript-eslint/typescript-eslint/issues/2728 is done
-        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
-        if (options && !options?.shell)
-        {
-            // Remove this when https://github.com/typescript-eslint/typescript-eslint/issues/2728 is done
+            if (!options?.cwd)
+            {
+                // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+                options.cwd = path.resolve(__dirname);
+            }
```

Comment:
nit Q: would this work?
```suggestion
            options.cwd ??= path.resolve(__dirname);
```